### PR TITLE
Restart k3s service unit on file change

### PR DESCRIPTION
**Pull request by pedrohdz**
The K3s service on the control node is not being restarted when there is a change.

### CODE REVIEW

**Comment by pedrohdz** [comment]
I THINK I know what I need to do to fix this.  I will try and get back to it later.

**Comment by pedrohdz** [comment]
@xanmanning 
So please keep in mind that I am an Ansible noob.  I'm not sure if my fix is legit, but it feels hacky to be honest.

I'm open to suggestions.  Let's see if the tests pass or course!

**Comment by xanmanning** [comment]
Hi @pedrohdz , I'll wait until the tests have run and either way I'll take a look at the implementation.

File: tasks/build/install-k3s.yml [hunk 1]
```diff
@@ -17,9 +17,17 @@
...
+  register: k3s_service_unit
 
 - meta: flush_handlers
 
+- name: Ensure k3s restarts after reload systemd, and if service unit file change on control plane
+  debug:
+    msg: Notify restart k3s
+  notify:
+    - restart k3s
+  when: k3s_service_unit.changed  # noqa 503
```
Comment (xanmanning) [inline tasks/build/install-k3s.yml:29]: Yes, this is a bit hacky. Still having a bit of a look and I might be able to come up with something, though it does seem to pass testing.

Comment (pedrohdz) [reply tasks/build/install-k3s.yml:29]: Thanks for taking a look.  I'm open to ideas.

**Comment by xanmanning** [comment]
Does this work for you @pedrohdz ? The issue with the service not restarting at the point required is due to the k3s binary not existing on the control node (it's downloaded but not symlinked to the right location). There's a bit of duplication but it would allow the service to start I believe.

**Comment by pedrohdz** [comment]
I think it will.  Should my hack be removed then?  I won't be able to test this until later this week, sorry.

**Comment by xanmanning** [comment]
Don't worry about when you can get round to testing, no rush. I pushed it to your branch for your review, if it doesn't work for you then feel free to revert it.

**Comment by pedrohdz** [comment]
Your fix works, and is a lot cleaner.

I rewrote the git history to rip out my hack and get rid of the noisy merge information.  If you want to pull the `control-node-restart-k3s` branch locally to get the changes, do a `git pull --force`.

Assuming my changes pass CI, this should be good to merge.

Thank you for taking the time!

**Comment by pedrohdz** [comment]
Tests seem to have passed but the GitHub PR is not showing it for some strange reason.

**Comment by xanmanning** [comment]
No worries, thanks for your assistance with this @pedrohdz - I shall merge it in now.

**Pull request merged**